### DISCUSSION TITLE
Importing reference spectra for Raman Mapping

### DISCUSSION THREAD

**Discussion by dimaria789** [Quasars/orange-spectroscopy#722] [Q&A] [answered]
Hello
I am trying to import the csv file and cannot create a spectrum from this file. Maybe someone knows a trick.
Cheers
Thanks
Maria

**Comment by borondics** [comment]
Hi Maria,

Could you upload the file somewhere? We can have a look and hopefully help you out.

Ferenc

**Reply by dimaria789** [reply]
Thanks a lot!!
Of course, let me know where, here is the file.
[fig 22-todorokte.csv](https://github.com/user-attachments/files/15808872/fig.22-todorokte.csv)

 I digitalized this spectrum from a paper, and i have my own Raman mapping. The idea was to "teach" the QUASAR to make a minerals mapping.

**Comment by borondics** [comment]
Hello!

Try to use this workflow:
[untitled.ows.zip](https://github.com/user-attachments/files/15810957/untitled.ows.zip)

You can use either the File or the CSV File import widget but they need to be configured correctly.
<img width="333" alt="image" src="https://github.com/Quasars/orange-spectroscopy/assets/6304794/53b4fd36-b45f-4657-8e05-472ec26d230c">


<img width="242" alt="image" src="https://github.com/Quasars/orange-spectroscopy/assets/6304794/1609bcba-1cff-41ed-a27f-fdf9d5431347">


Or: 

<img width="600" alt="image" src="https://github.com/Quasars/orange-spectroscopy/assets/6304794/61b963db-0c22-487c-9ce5-09e5d23ad305">


<img width="242" alt="image" src="https://github.com/Quasars/orange-spectroscopy/assets/6304794/ba2c74cb-4470-416d-9c5a-314f1c9b8539">

I hope this helps.

**Comment by dimaria789** [comment]
Thanks! I did  the same, and tried to transpose the column, but then the spectral data (the wavelength) is getting lost, you can see in the  figure, the x-axis is from 1 to 200 instead of the wavelength (ca. 800 to 1800 cm-1). Thanks! Maria

**Comment by borondics** [comment]
I think you might need to click the "Remove redundant instance".

**Reply by dimaria789** [reply]
It is still not working, not giving the wavelength

**Reply by borondics** [reply, marked as answer]
You are right... Something seems funny here. @markotoplak, could you have a look?

**Reply by dimaria789** [reply]
Thanks!

**Comment by dimaria789** [comment]
Hi Ferenc
it is clicked and still I am getting 
X axis 1 to 200, not the numeric data in the features, theres is a note: ￼"Values are not unique.
To avoid multiple features with the same name, values 
of 'Wave' have been augmented with indices." 
Sorry, for the stupid problem.
Thanks
Maria

**Comment by dimaria789** [comment]
Hey
Any advise, I still not getting the wave-numbers, but just counting on the X-axis
Thanks a lot
Maria

Maria Dittrich, PhD

Professor

Earth Sciences Department

University of Toronto

22 Ursula Franklin Street, M5S 3B1, Toronto

Department of Physical and Environmental Sciences

University of Toronto Scarborough

1265 Military Trail, M1C 1A4, Toronto
ON,  Canada










________________________________

**Reply by markotoplak** [reply]
Maria, I tried the second option that @borondics showed above. I used the "File" for reading and then "Transpose", set to "From variable" to "Wave" and "Remove redundant instance" checked. Are you sure you did both? Or does what I am getting seem somewhat incorect?

If this does not help, can you show me a screenshot of related components? Here is mine.

![a](https://github.com/Quasars/orange-spectroscopy/assets/552182/cc561231-ec0c-4873-8792-3d5c90dd5f34)

**Reply by dimaria789** [reply]
Can you please check the spectra afterwards. This is what is not working. Thanks a lot!

**Reply by markotoplak** [reply]
Ok, now I finally understand the issue. Spectra can not understand your wavenumbers spectrum because some wavenumbers are repeating. Your data contains 734.2035432 and 738.4409664 twice.  To work around it you could remove them with select columns after Transpose, like this. This is actually why the Transpose component has a warning triangle.

![image](https://github.com/Quasars/orange-spectroscopy/assets/552182/1790783c-667f-432b-8657-b8809d8bfd61)

I know this is a lousy workaround but can not think of anything nicer.


Now that we understand why this is happening: is your data intentionally like this? If so, what do you aim to work on/display/analyze? I'd like to understand more about what you planned doing, because it seems that some components that could handle data like yours could be generally useful.


Furthermore, how did you imagine this spectrum (where some wavenumbers repeat) to be displayed? As an average on repeated wavenumbers? Or as multiple possibilities in plot, somehow marked by dots?

**Comment by dimaria789** [comment]
Mark, yes, I tried it, but when I go to the spectra, it does not show the wavelengths, but just count numbers 1-200, on the X-axis.
I followed your instructions, but now I am getting this

![Screenshot 2024-07-08 at 10 27 33 AM](https://github.com/Quasars/orange-spectroscopy/assets/164642478/bd1a14fe-118a-49cf-9728-0117a0cf4b24)
![Screenshot 2024-07-08 at 10 23 41 AM](https://github.com/Quasars/orange-spectroscopy/assets/164642478/7a3e7e9f-2b96-49aa-91fc-91f29ff5183e)

**Reply by markotoplak** [reply]
Could you show me what "File (1)" settings look like? It seems that "Intensity" role is set to "meta" in your case which it should no be. Can you try changing it (click on a cell and change them). Mine looks like this.

![image](https://github.com/Quasars/orange-spectroscopy/assets/552182/eeca5555-4721-4058-b5fd-55b80f55226f)

**Comment by dimaria789** [comment]
when I did CSV import:
![Screenshot 2024-07-08 at 10 32 11 AM](https://github.com/Quasars/orange-spectroscopy/assets/164642478/69ca2960-138e-44f1-8a04-5b36822fd503)

I am getting 1-220 for the X-values, 
![Screenshot 2024-07-08 at 10 33 21 AM](https://github.com/Quasars/orange-spectroscopy/assets/164642478/c4c57f77-78c4-4e90-8d8e-f549da548ae2)
and not the wavelengths:

![Screenshot 2024-07-08 at 10 34 18 AM](https://github.com/Quasars/orange-spectroscopy/assets/164642478/218344e3-7f81-4e6e-b979-edf49c278495)

**Reply by markotoplak** [reply]
Ok, now I finally understand the issue. Spectra can not understand your wavenumbers spectrum because some wavenumbers are repeating. Your data contains 734.2035432 and 738.4409664 twice.  To work around it you could remove them with select columns after Transpose, like this. This is actually why the Transpose component has a warning triangle.

![image](https://github.com/Quasars/orange-spectroscopy/assets/552182/1790783c-667f-432b-8657-b8809d8bfd61)

I know this is a lousy workaround but can not think of anything nicer.


Now that we understand why this is happening: is your data intentionally like this? If so, what do you aim to work on/display/analyze? I'd like to understand more about what you planned doing, because it seems that some components that could handle data like yours could be generally useful.


Furthermore, how did you imagine this spectrum (where some wavenumbers repeat) to be displayed? As an average on repeated wavenumbers? Or as multiple possibilities in plot, somehow marked by dots?

**Comment by dimaria789** [comment]
THANKS! It works now!
Yes, I see the warning triangle. I ticked the remove the redundant instance.
How did you find the doubles?

No, BUT I digitalized it with origin, so my mistake!
The idea is to "learn" the references spectra for the Raman mapping on Mn-Fe nodules from deep ocean, to see how Ni Co etc are incorporated.

Maria Dittrich, PhD

Professor

Earth Sciences Department

University of Toronto

22 Ursula Franklin Street, M5S 3B1, Toronto

Department of Physical and Environmental Sciences

University of Toronto Scarborough

1265 Military Trail, M1C 1A4, Toronto
ON,  Canada










________________________________
